src/components/SearchBarPopup/SearchBarPopup.tsx

```diff
@@ -0,0 +1,159 @@
+        new Promise<Array<Destination>>(function (resolve, reject) {
+            /** TODO: Modify logic here to make HTTP call to backend for fetching search results.
+             *  Update DB if the previous result list has been changed.
+             */
+
+            let results: Destination[] = [{id: 1, name: "Place 1", address:"City, County", isAdded: false}, 
+                {id: 2, name: "Place 2", address: "City, County", isAdded: false }, 
+                {id: 3, name: "Place 3", address: "City, County", isAdded: false }];
+            resolve(results);
+        })
+            .then((results: Destination[]) => {
+                this.setState({ resultsOpen: true, loading: false, searchKey: searchInput });
+
+                // Todo: this needs to be restored afterwards! 
+                // this.setState({ resultsOpen: true, results: results, loading: false });
+            })
+            .catch((err) => {
+                this.setState({
+                    resultsOpen: false,
+                    loading: false,
+                    errMsg: err.toString(),
+                });
+            });
+    }
+
+    private addRemovePlace (id: number) {
+        const elementsIndex = this.state.results.findIndex(element => element.id === id);
+        let newResults = [...this.state.results];
+        newResults[elementsIndex] = { ...newResults[elementsIndex], isAdded: !newResults[elementsIndex].isAdded}
+        this.setState({
+            results: newResults,
+            resultsOpen: true
+        })
+    }
+
+    componentDidMount(): void {
+        document.addEventListener("mousedown", this.handleClickOutside, false);
+    }
+
+    componentWillUnmount(): void {
+        document.removeEventListener("mousedown", this.handleClickOutside, false);
+    }
+
+    private handleClickOutside = (event: MouseEvent) => {
+        if (this.container.current && event.target instanceof Node) {
+            if (this.container.current.contains(event.target)) {
+                return;
+            }  else {
+                // TODO: backend, send the updated result list to the backend
+            }
+        }
+    };
+    
+    private renderSearchResults() {
+        const { results } = this.state;
+        // console.log(this.state.resultsOpen)
+        if (this.state.resultsOpen && _.isArray(results) && !_.isEmpty(results)) {
+            return (
+                <div className="results-container">
+                    <ul className="results-list">
+                        {results.map((result) => (
+                            // temporarily using result as key. Change to some id later
+                            // TODO: replace with valid href
+                            // TODO:<a href> remount the entire component, fix when connecting to backend.
+                            <li className="results-entry" key={result["name"]} onClick={() =>this.addRemovePlace(result["id"])}>
+                                <a href="placeholder">
```

Comment:
LOL i just realized.... we don't even need the link tag cuz we're redirecting them to another page when they click on the search result. Then we don't have the problem of component re-construction on click either.